src/internet_identity/src/main.rs

```diff
@@ -84,13 +85,20 @@ enum KeyType {
     SeedPhrase,
 }
 
+#[derive(Clone, Debug, CandidType, Deserialize, PartialEq)]
+enum Tag {
+    #[serde(rename = "protected")]
+    Protected,
+}
+
 #[derive(Clone, Debug, CandidType, Deserialize)]
 struct DeviceData {
     pubkey: DeviceKey,
     alias: String,
     credential_id: Option<CredentialId>,
     purpose: Purpose,
     key_type: KeyType,
+    tags: Option<Vec<Tag>>
```

Comment:
Does this need to be an `Option` due to some backwards compatibility concern?

Thinking `Vec<Tag>` would be sufficient otherwise.

Reply:
You are right. But I think in case of rollback Optional field is more convenient.